### PR TITLE
Feature/add meson support

### PR DESCRIPTION
**Pull request by mleivaf**
Adding Meson build system support.

**Commands:**

**Configuration.**
```
mkdir buildmeson
~/.local/bin/meson buildmeson/ --prefix=/usr
```
or
`~/.local/bin/meson buildmeson/ --prefix=/usr -Denable-gtk-doc=false`
**Build.**
```
ninja -C buildmeson
```
**Run tests.**
```
ninja -C buildmeson test
```
**Installation.**
```
sudo ninja -C buildmeson install
```
Specify a different installation directory
```
DESTDIR=~/meson/interpipes-meson-install ninja -C buildmeson install
```

### CODE REVIEW

File: docs/meson.build [hunk 1]
```diff
@@ -0,0 +1,10 @@
+docconf = configuration_data()
+docconf.set('GST_API_VERSION', apiversion)
+docconf.set('VERSION', gst_interpipes_version)
+docconf.set('PLUGINDIR', '@0@/lib/gstreamer-1.0'.format(get_option('prefix')))
```
Comment (michaelgruner) [inline docs/meson.build]: This should be `libdir`. In Ubuntu, for example, `prefix/lib != (libdir=/usr/lib/x86_64-linux-gnu)`.

File: docs/meson.build [hunk 2]
```diff
@@ -0,0 +1,10 @@
+docconf = configuration_data()
+docconf.set('GST_API_VERSION', apiversion)
```
Comment (michaelgruner) [inline docs/meson.build]: Are these still necessary, even after removing the `xml.in`?

File: docs/plugins/meson.build [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+types = configure_file(input : 'gst-interpipe.types',
+  output : 'gst-interpipe.types',
+  copy : true)
+
+version_conf = configuration_data()
+version_conf.set('PACKAGE_VERSION', meson.project_version())
+version_conf.set('GST_API_VERSION', apiversion)
+version_conf.set('PACKAGE_NAME', 'GstInterpipe')
+version_conf.set('PACKAGE_URL', pkg_interpipes_url)
+configure_file(input : 'gst-interpipe-docs.xml.in',
```
Comment (michaelgruner) [inline docs/plugins/meson.build]: This is no longer necessary

File: meson.build [hunk 4]
```diff
@@ -0,0 +1,174 @@
...
+# Define compiler args and include directories
+gst_c_args = ['-DHAVE_CONFIG_H']
+configinc = include_directories('.')
+
+# Define gstreamer API version
+apiversion = '1.0'
+
+# Define installation directories
+prefix = get_option('prefix')
+plugin_install_dir = join_paths(get_option('prefix'), get_option('libdir'), 'gstreamer-1.0')
```
Comment (michaelgruner) [inline meson.build]: Is this okay? doesn't `libdir` already absorb `prefix`?

File: meson_options.txt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+# Feature options
+option('tests', type : 'feature', value : 'auto', yield : true, description : 'Enable tests')
```
Comment (michaelgruner) [inline meson_options.txt]: Unless the gtk-doc module already does so, please add an option to disable doc building.

File: docs/meson.build (hunk 1 above)
Comment (mleivaf) [reply docs/meson.build]: Yes, we should use libdir

File: docs/meson.build (hunk 2 above)
Comment (mleivaf) [reply docs/meson.build]: gnome.gtkdoc parameters were changed in order to use  the `gst-interpipe-docs.xml` generated, now the procedure is equivalent to the procedure applied by autotools, therefore this process is required.

File: docs/plugins/meson.build (hunk 3 above)
Comment (mleivaf) [reply docs/plugins/meson.build]: gnome.gtkdoc parameters were changed in order to use the gst-interpipe-docs.xml generated,  therefore this process is required now.

File: meson.build (hunk 4 above)
Comment (mleivaf) [reply meson.build]: Yes, we should use libdir

File: meson_options.txt (hunk 5 above)
Comment (mleivaf) [reply meson_options.txt]: The option enable-gtk-doc was added.

**Pull request merged**